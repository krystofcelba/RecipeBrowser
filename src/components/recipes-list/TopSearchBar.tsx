import React from 'react';
import { StyleSheet, View } from 'react-native';
import { connect } from 'react-redux';

import { Colors, StylesConstants } from 'src/assets/constants';
import i18n from 'src/assets/i18n';
import { SearchBar } from 'react-native-elements';
import { actionCreators } from 'src/redux/reducers/ui';
import { getRecipesSearchInputText } from 'src/redux/selectors';

interface Props {
  updateRecipesSearchInputText: typeof actionCreators.updateRecipesSearchInputText;
  showClearIcon?: boolean;
}

const TopSearchBar = ({ updateRecipesSearchInputText, showClearIcon = false }: Props) => (
  <View style={styles.container}>
    <SearchBar
      lightTheme
      round
      containerStyle={styles.searchBar}
      inputStyle={styles.searchBarInput}
      onChangeText={updateRecipesSearchInputText}
      placeholder={i18n.t('search')}
      placeholderTextColor={Colors.searchBarPlaceholderTextColor}
      selectionColor={Colors.searchBarTextColor}
      icon={{ color: Colors.searchBarPlaceholderTextColor, name: 'search' }}
      clearIcon={showClearIcon ? { color: Colors.searchBarPlaceholderTextColor, name: 'clear' } : undefined}
    />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: Colors.primaryColor,
  },
  searchBar: {
    flex: 1,
    backgroundColor: Colors.primaryColor,
    borderBottomWidth: 0,
    borderTopWidth: 0,
  },
  searchBarInput: {
    backgroundColor: Colors.darkPrimaryColor,
    color: Colors.searchBarTextColor,
    fontFamily: StylesConstants.fontFamily,
  },
});

const mapStateToProps = state => ({ showClearIcon: getRecipesSearchInputText(state).length });

const mapDispatchToProps = {
  updateRecipesSearchInputText: actionCreators.updateRecipesSearchInputText,
};

export default connect(mapStateToProps, mapDispatchToProps)(TopSearchBar);
