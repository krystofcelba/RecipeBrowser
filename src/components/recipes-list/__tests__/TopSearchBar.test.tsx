import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import TopSearchBar from '../TopSearchBar';

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {
  ui: {
    recipesList: { searchInput: { text: '' } },
  },
};

const stateWithSomeInputText = {
  ui: {
    recipesList: { searchInput: { text: 'grape' } },
  },
};

describe('TopSearchBar component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<TopSearchBar />, { context: { store: mockStore(initialState) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders as expected with clear icon when there is some text', () => {
    const wrapper = shallow(<TopSearchBar />, { context: { store: mockStore(stateWithSomeInputText) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
