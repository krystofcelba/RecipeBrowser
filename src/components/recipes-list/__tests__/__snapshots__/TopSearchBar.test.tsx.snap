// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSearchBar component renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#81b640",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Search
    containerStyle={
      Object {
        "backgroundColor": "#81b640",
        "borderBottomWidth": 0,
        "borderTopWidth": 0,
        "flex": 1,
      }
    }
    icon={
      Object {
        "color": "rgba(255, 255, 255, 0.3)",
        "name": "search",
      }
    }
    inputStyle={
      Object {
        "backgroundColor": "#508607",
        "color": "rgba(255, 255, 255, 1.0)",
        "fontFamily": "Avenir",
      }
    }
    lightTheme={true}
    loadingIcon={Object {}}
    noIcon={false}
    onChangeText={[Function]}
    placeholder="Search"
    placeholderTextColor="rgba(255, 255, 255, 0.3)"
    round={true}
    selectionColor="rgba(255, 255, 255, 1.0)"
    showLoadingIcon={false}
  />
</View>
`;

exports[`TopSearchBar component renders as expected with clear icon when there is some text 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#81b640",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Search
    clearIcon={
      Object {
        "color": "rgba(255, 255, 255, 0.3)",
        "name": "clear",
      }
    }
    containerStyle={
      Object {
        "backgroundColor": "#81b640",
        "borderBottomWidth": 0,
        "borderTopWidth": 0,
        "flex": 1,
      }
    }
    icon={
      Object {
        "color": "rgba(255, 255, 255, 0.3)",
        "name": "search",
      }
    }
    inputStyle={
      Object {
        "backgroundColor": "#508607",
        "color": "rgba(255, 255, 255, 1.0)",
        "fontFamily": "Avenir",
      }
    }
    lightTheme={true}
    loadingIcon={Object {}}
    noIcon={false}
    onChangeText={[Function]}
    placeholder="Search"
    placeholderTextColor="rgba(255, 255, 255, 0.3)"
    round={true}
    selectionColor="rgba(255, 255, 255, 1.0)"
    showLoadingIcon={false}
  />
</View>
`;
